price-feeder/oracle/oracle.go

```diff
@@ -147,17 +147,25 @@ func (o *Oracle) SetPrices() error {
 	g := new(errgroup.Group)
 	mtx := new(sync.Mutex)
 	providerPrices := make(map[string]map[string]provider.TickerPrice)
+	requiredRates := []string{}
 
 	for providerName, currencyPairs := range o.providerPairs {
 		providerName := providerName
 		currencyPairs := currencyPairs
 
 		priceProvider := o.getOrSetProvider(providerName)
 
+		for _, pair := range currencyPairs {
+			if !Contains(requiredRates, pair.Base) {
+				requiredRates = append(requiredRates, pair.Base)
+			}
+		}
+
 		g.Go(func() error {
 			prices, err := priceProvider.GetTickerPrices(currencyPairs...)
 			if err != nil {
-				return err
+				o.logger.Debug().Msg("getting ticker price failed")
+				return nil
```

Comment:
I think we still want to return an error here. We have no need to continue on with the function closure if this call fails. Correct me if I'm wrong though.